tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_opcodes.py

```diff
@@ -375,259 +416,1614 @@ def test_bal_extcodesize_and_oog(
 ) -> None:
-    """Ensure BAL handles CALL and OOG during CALL appropriately."""
-    alice = pre.fund_eoa()
-    bob = pre.fund_eoa()
+    """CALL without 7702 delegation - test successful execution."""
     gas_costs = fork.gas_costs()
+    alice = pre.fund_eoa()
 
-    # Create contract that attempts to call Bob
-    call_contract_code = Bytecode(
-        Op.PUSH1(0)  # retSize
-        + Op.PUSH1(0)  # retOffset
-        + Op.PUSH1(0)  # argsSize
-        + Op.PUSH1(0)  # argsOffset
-        + Op.PUSH1(0)  # value
-        + Op.PUSH20(bob)  # address
-        + Op.PUSH2(0xFFFF)  # gas (provide enough for the call)
-        + Op.CALL  # Call (cold account access)
-        + Op.STOP
+    target = (
+        pre.empty_account()
+        if target_is_empty
+        else pre.deploy_contract(code=Op.STOP)
     )
 
-    call_contract = pre.deploy_contract(code=call_contract_code)
+    # warmup code if needed
+    warmup_code = Bytecode()
+    if target_is_warm:
+        warmup_code = Op.POP(Op.EXTCODESIZE(target))
 
-    intrinsic_gas_calculator = fork.transaction_intrinsic_cost_calculator()
-    intrinsic_gas_cost = intrinsic_gas_calculator()
+    # memory expansion / no expansion
+    ret_size = 32 if memory_expansion else 0
 
-    # Costs:
-    # - 7 PUSH operations = G_VERY_LOW * 7
-    # - CALL cold = G_COLD_ACCOUNT_ACCESS (minimum for account access)
-    push_cost = gas_costs.G_VERY_LOW * 7
-    call_cold_cost = gas_costs.G_COLD_ACCOUNT_ACCESS
-    tx_gas_limit = intrinsic_gas_cost + push_cost + call_cold_cost
-
-    if fails_at_call:
-        # subtract 1 gas to ensure OOG at CALL
-        tx_gas_limit -= 1
+    # caller contract
+    call_code = warmup_code + Op.CALL(
+        gas=0, address=target, value=value, ret_size=ret_size
+    )
+    caller = pre.deploy_contract(code=call_code, balance=value)
 
-    tx = Transaction(
-        sender=alice,
-        to=call_contract,
-        gas_limit=tx_gas_limit,
+    intrinsic_cost = fork.transaction_intrinsic_cost_calculator()()
+
+    # Warmup cost: bytecode (PUSH20 + POP) + state access (EXTCODESIZE cold)
+    warmup_total = (
+        (call_warmup_bytecode_cost + gas_costs.G_COLD_ACCOUNT_ACCESS)
+        if target_is_warm
+        else 0
     )
 
-    block = Block(
-        txs=[tx],
-        expected_block_access_list=BlockAccessListExpectation(
-            account_expectations={
-                call_contract: BalAccountExpectation.empty(),
-                # Bob should only appear if CALL succeeded
-                **(
-                    {bob: None}
-                    if fails_at_call
-                    else {bob: BalAccountExpectation.empty()}
-                ),
-            }
-        ),
+    # Bytecode cost: 7 pushes for Op.CALL + warmup
+    bytecode_cost = gas_costs.G_VERY_LOW * 7 + warmup_total
+
+    # Access cost for CALL
+    access_cost = (
+        gas_costs.G_WARM_ACCOUNT_ACCESS
+        if target_is_warm
+        else gas_costs.G_COLD_ACCOUNT_ACCESS
```

Comment:
Hmm I believe if not `target_is_warm` `G_COLD_ACCOUNT_ACCESS` is added twice here? See line 466

Reply:
Yup stale 👍 I should permalink those lines instead of reffing them like this 😓